t/09_qesapdeployment.t

```diff
@@ -789,4 +789,96 @@ subtest '[qesap_prepare_env::qesap_create_aws_config] not solved template and va
     like $contents[0], qr/region = eu-central-1/, "Expected region eu-central-1 is in the config file";
 };
 
+subtest '[qesap_is_job_finished]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    my @results = ();
+    $qesap->redefine(script_output => sub {
+            if ($_[0] =~ /100000/) { return "not json"; }
+            if ($_[0] =~ /200000/) { return "{\"job\":{\"state\":\"donaldduck\"}}"; }
+            if ($_[0] =~ /300000/) { return "{\"job\":{\"state\":\"running\"}}"; }
+    });
+
+    $qesap->redefine(get_required_var => sub { return ''; });
```

Comment:
potentially you do not need to do it. In place of that you can call `set_var('SOME_SETTING', 'somevalue')` and before the assert section  `set_var('SOME_SETTING', undef)`

Reply:
Yes, but that would be more complex to essentially achieve the same thing, no? Since we don't want to check the var's value in some way (it's the openqa server hostname, and it has 0 interest in the test context) , we can just make `get_required_var` return the empty string